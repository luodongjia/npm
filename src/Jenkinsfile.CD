def label = "worker-${UUID.randomUUID().toString()}"

podTemplate(label: label, cloud: 'kubernetes',
    containers: [
        containerTemplate(name: 'wtctl', image: 'worktile/wtctl:1.1', ttyEnabled: true, command: 'cat')
    ],
    volumes: [
        hostPathVolume(mountPath: '/var/run/docker.sock', hostPath: '/var/run/docker.sock'),
        hostPathVolume(mountPath: '/tmp/cache', hostPath: '/tmp/cache/wt-rd-pipeline'),
        hostPathVolume(mountPath: '/root/.ssh', hostPath: '/root/.ssh')
    ]
) {
    node(label) {

        def scmVars = checkout scm

        def branch = scmVars.GIT_BRANCH

        stage('Deploy Backend Test Environment') {
            script {
                if (branch == "origin/master" || branch == "master") {
                    echo 'Ready to deploy environment'

                    container("wtctl") {
                        sh "wtctl --test --version"
                    }
                }
                else {
                    echo 'Skip the deployment'
                }
            }
        }
    }
 }