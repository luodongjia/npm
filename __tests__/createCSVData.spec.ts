import { testTableItems } from './shared/data'
import { createCSVData } from '../src/converters'

it('createCSVData works correctly', () => {
  expect(createCSVData(testTableItems)).toMatchSnapshot()
})
