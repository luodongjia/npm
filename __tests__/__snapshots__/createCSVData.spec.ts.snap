// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCSVData works correctly 1`] = `
"greeting,foo,bar,a,b,c,d,e,f
\\"hello\\",\\"\\",\\"\\",\\"\\",\\"\\",\\"\\",\\"\\",\\"\\",\\"\\"
\\"\\",\\"foo\\",\\"bar\\",\\"\\",\\"\\",\\"\\",\\"\\",\\"\\",\\"\\"
\\"\\",\\"\\",\\"\\",\\"100\\",\\"[1,2,3]\\",\\"{\\"\\"x\\"\\":200,\\"\\"y\\"\\":300}\\",\\"D\\",\\"null\\",\\"\\""
`;
